.travis.yml

```diff
@@ -146,16 +141,23 @@ script:
   - echo 'package criterion-examples' >> cabal.project
   - "echo '  ghc-options: -Werror' >> cabal.project"
   - touch cabal.project.local
-  - "if ! $NOINSTALLEDCONSTRAINTS; then for pkg in $($HCPKG list --simple-output); do echo $pkg  | grep -vw -- criterion | grep -vw -- criterion-measurement | grep -vw -- criterion-examples | sed 's/^/constraints: /' | sed 's/-[^-]*$/ installed/' >> cabal.project.local; done; fi"
+  - "for pkg in $($HCPKG list --simple-output); do echo $pkg | sed 's/-[^-]*$//' | grep -vE -- '^(criterion|criterion-examples|criterion-measurement)$' | sed 's/^/constraints: /' | sed 's/$/ installed/' >> cabal.project.local; done"
   - cat cabal.project || true
   - cat cabal.project.local || true
 
   # build & run tests, build benchmarks
-  - cabal new-build -w ${HC} ${CABALFLAGS} ${TEST} ${BENCH} all
-  - if [ "x$TEST" = "x--enable-tests" ]; then cabal new-test -w ${HC} ${CABALFLAGS} ${TEST} ${BENCH} all; fi
+  - cabal new-build -w ${HC} ${TEST} ${BENCH} all
+  - if [ "x$TEST" = "x--enable-tests" ]; then cabal new-test -w ${HC} ${TEST} ${BENCH} all; fi
 
   # haddock
-  - if $HADDOCK; then cabal new-haddock -w ${HC} ${CABALFLAGS} ${TEST} ${BENCH} all; else echo "Skipping haddock generation";fi
+  - cabal new-haddock -w ${HC} ${TEST} ${BENCH} all
+
+  # Constraint sets
+  - rm -rf cabal.project.local
+
+  # Constraint set embed-files
+  - if [ $HCNUMVER -ge 80600 ] && [ $HCNUMVER -lt 80700 ] ; then cabal new-test  -w ${HC} --enable-tests  --enable-benchmarks  --constraint='criterion +embed-data-files' all ; fi
```

Comment:
Shouldn't `--enable-tests  --enable-benchmarks` be `${TEST} ${BENCH}`?

Reply:
Also, the mechanism for accomplishing this seems a bit crude. I would have expected something closer to this:

```bash
do
  cabal new-build -w ${HC} ${TEST} ${BENCH} --constraint='criterion +embed-data-files' all
  cabal new-test  -w ${HC} ${TEST} ${BENCH} --constraint='criterion +embed-data-files' all
done
```

Not only does this more closely mirror what we're doing in the non-`embed-data-files` case, but it would also make it easier to add or drop lines as necessary. (For instance, I would have hoped that we could run the Haddocks with `embed-data-files` enabled as well.)

Reply:
I'll go with this for now, feel free to make a follow-up PR in `haskell-ci`.

Reply:
I mean... you're still actively working on haskell-CI/haskell-ci#212, right? It can't be that hard to just add this from the get-go.

Reply:
I spent most of the day (which is not work time) on it. Sorry, have to draw a line somewhere.

Reply:
Sure, that's fine. But do be aware that I'm not going to merge this until we can reach feature parity with what the current `.travis.yml` does.

Reply:
This PR is a demo, to check what travis says :) you can can close this after I get green from travis.

Reply:
No worries! I do want to land something analogous to this at some point, so I'd be fine with keeping this open for testing further changes. (I'll mark it as a WIP in the meantime.)